package/MDAnalysis/analysis/density.py

```diff
@@ -383,7 +384,18 @@ def __init__(self, atomgroup, delta=1.0,
         self._ydim = ydim
         self._zdim = zdim
 
+    def _checkinput(self):
+        if len(self._atomgroup) == 0:
```

Comment:
It's worth revisiting the original issue and checking what we need here. It's correct that we want to avoid building a grid on an empty selection when we aren't using a user defined grid, but when we are, we want to be able to let users pass an empty selection.

It's worth considering where in the code the edge case is failing (I'll let you work it out, but I believe it might be `smin` and `smax`), and how you could incorporate such a check within this context.

As @hmacdope mentioned, it's worth looking to apply a test driven development approach here. Write some tests that would check the new behaviour of the code post fix, and then use those as ways to verify if your code changes are indeed addressing the issue.

Reply:
Hi, Thanks for all the suggestions! I will try to clean all this up. I think I need to approach this issue another way. I'll try to catch this issue from the smin,smax step.